openshift/jag-csrs.yml

```diff
@@ -0,0 +1,164 @@
+  metadata:
+    name: ${COMPONENT_NAME}
+    namespace: ${OC_NAMESPACE}-${OC_ENV}
+      
+  spec:
+    completionDeadlineSeconds: 1200
+    replicas: 1
+    selector:
+      name: ${COMPONENT_NAME}
+    strategy:
+      type: Rolling
+      maxSurge: 50%
+      maxUnavailable: 0
+    template:
+      metadata:
+        labels:
+          name: ${COMPONENT_NAME}
+      spec:
+        containers:
+          # due to permission issue not using tools space; dev is pulling from dev
+          - image: >-
+              image-registry.openshift-image-registry.svc:5000/${OC_NAMESPACE}-${OC_ENV}/${APP_NAME}:${OC_ENV}
+            name: ${COMPONENT_NAME} 
+            imagePullPolicy: Always
+            ports:
+              - containerPort: 8080
+            env:
+            - name: SPLUNK_HTTP_URL
+              valueFrom:
+                secretKeyRef:
+                  name: ${SPLUNK_CONFIG_NAME}
+                  key: SPLUNK_HTTP_URL
+            - name: SPLUNK_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: ${SPLUNK_CONFIG_NAME}
+                  key: SPLUNK_TOKEN
+            - name: SPLUNK_INDEX
+              valueFrom:
+                secretKeyRef:
+                  name: ${SPLUNK_CONFIG_NAME}
+                  key: SPLUNK_INDEX
+            - name: BCEID_URL
+              value: ${BCEID_URL}
+            resources:
+              limits:
+                cpu: 250m
+                memory: 1000Mi
+              requests:
+                cpu: 125m
+                memory: 512Mi
+            # Took away readiness and liveness probe until those end points are ready
+    triggers:
+      - type: ConfigChange
+      - type: ImageChange
+        imageChangeParams:
+          automatic: true
+          containerNames:
+            - ${COMPONENT_NAME}
+          from:
+            kind: ImageStreamTag
+            namespace: ${OC_NAMESPACE}-${OC_ENV} #do not have permission to leverage tools
+            name: ${COMPONENT_NAME}:${OC_ENV}
+  status: {}
+
+### JAG-csrs Service ###
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: ${COMPONENT_NAME}
+    namespace: ${OC_NAMESPACE}-${OC_ENV}
+  spec:
+    ports:
+      - name: http
+        protocol: TCP
+        port: 80
+        targetPort: 8080
+    selector:
+      name: ${COMPONENT_NAME}
+    sessionAffinity: None
+    type: ClusterIP
+
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    name: ${ROUTE_PREFIX}
+    namespace: ${OC_NAMESPACE}-${OC_ENV}
+  annotations:
+    # if we do not use sticky-session, disable_cookies needs to be true
+    haproxy.router.openshift.io/disable_cookies: "true"
+    # balance algorithm
+    haproxy.router.openshift.io/balance: roundrobin
+  spec:
+    host: ${ROUTE_PREFIX}-${OC_NAMESPACE}-${OC_ENV}.${DNS}
+    to:
+      kind: Service
+      name: ${COMPONENT_NAME}
+    port:
+      targetPort: http
+    tls:
+      termination: edge
+      insecureEdgeTerminationPolicy: None
```

Comment:
This should probably be,

```yaml
insecureEdgeTerminationPolicy: Redirect
```

to ensure any http:// calls automatically get redirected to https://

Reply:
Redirection is a good idea in this scenario. Thank you!

Reply:
Commit https://github.com/bcgov/jag-csrs-portal-public/commit/bb99a7d3aa60400c6295ec14e2e00248386f4d74 
Thanks!